netty/src/main/java/io/grpc/netty/NettyClientStream.java

```diff
@@ -92,6 +92,9 @@ class NettyClientStream extends AbstractClientStream {
     this.authority = checkNotNull(authority, "authority");
     this.scheme = checkNotNull(scheme, "scheme");
     this.userAgent = userAgent;
+    if (callOptions.getOnReadyThreshold() != null) {
```

Comment:
If you moved this code to `AbstractClientStream` then you wouldn't need it in each transport individually. `CallOptions` is already passed to the `AbstractClientStream` constructor. You could also get rid of `setOnReadyThreshold()` from `Stream` and all the client-side forwarding classes. We'd still need something for server-side, but there are fewer classes involved there.

Reply:
I went down this route now and ran into problems. Some subclasses of AbstractClientStream construct the transport in their constructors after the super-class constructor runs, so transportState() is null during the superclass constructor.

We could defer actually setting it on the transportState() until AbstractClientStream#start() is run, assuming that must be run and doesn't get overridden.